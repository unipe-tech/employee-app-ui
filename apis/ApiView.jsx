import React from "react";
import { Button } from "@react-native-material/core";
import TextButton from "../components/atoms/TextButton";

const ApiView = (props) => {
  const { disabled, goForFetch, loading, style, title } = props;

  return (
    <TextButton
      label={loading ? "Verifying" : "Continue"}
      uppercase={false}
      type="solid"
      color="#4E46F1"
      style={style}
      disabled={loading || disabled}
      onPress={() => {
        goForFetch();
      }}
    />
  );
};

export default ApiView;
