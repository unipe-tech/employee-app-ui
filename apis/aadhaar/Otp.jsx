import { useEffect, useState } from "react";
import { Alert } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { OG_API_KEY } from "@env";
import { useNavigation } from "@react-navigation/core";
import Analytics from "appcenter-analytics";

import PrimaryButton from "../../components/PrimaryButton";
import { aadhaarBackendPush } from "../../helpers/BackendPush";
import { KYC_AADHAAR_GENERATE_OTP_API_URL } from "../../services/constants";
import {
  addSubmitOTPtxnId,
  addVerifyMsg,
  addVerifyStatus,
  addVerifyTimestamp,
} from "../../store/slices/aadhaarSlice";
import { resetTimer } from "../../store/slices/timerSlice";

const AadhaarOtpApi = (props) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const [loading, setLoading] = useState(false);
  const [backendPush, setBackendPush] = useState(false);

  const id = useSelector((state) => state.auth.id);
  const aadhaarSlice = useSelector((state) => state.aadhaar);
  const [submitOTPtxnId, setSubmitOTPtxnId] = useState(
    aadhaarSlice?.submitOTPtxnId
  );
  const [verifyMsg, setVerifyMsg] = useState(aadhaarSlice?.verifyMsg);
  const [verifyStatus, setVerifyStatus] = useState(aadhaarSlice?.verifyStatus);
  const [verifyTimestamp, setVerifyTimestamp] = useState(
    aadhaarSlice?.verifyTimestamp
  );

  useEffect(() => {
    dispatch(addSubmitOTPtxnId(submitOTPtxnId));
  }, [submitOTPtxnId]);

  useEffect(() => {
    dispatch(addVerifyMsg(verifyMsg));
  }, [verifyMsg]);

  useEffect(() => {
    dispatch(addVerifyStatus(verifyStatus));
  }, [verifyStatus]);

  useEffect(() => {
    dispatch(addVerifyTimestamp(verifyTimestamp));
  }, [verifyTimestamp]);

  useEffect(() => {
    console.log(
      "KYC_AADHAAR_GENERATE_OTP_API_URL: ",
      KYC_AADHAAR_GENERATE_OTP_API_URL
    );
    console.log("AadhaarOtpApi aadhaarSlice: ", aadhaarSlice);
    if (backendPush) {
      aadhaarBackendPush({
        id,
        data: aadhaarSlice?.data,
        number: aadhaarSlice?.number,
        verifyMsg,
        verifyStatus,
        verifyTimestamp,
      });
      setBackendPush(false);
      setLoading(false);
    }
  }, [backendPush]);

  const goForFetch = () => {
    setLoading(true);

    const options = {
      method: "POST",
      headers: {
        "X-Auth-Type": "API-Key",
        "X-API-Key": OG_API_KEY,
        "Content-Type": "application/json",
      },
      body: JSON.stringify(props.data),
    };

    fetch(KYC_AADHAAR_GENERATE_OTP_API_URL, options)
      .then((response) => response.json())
      .then((responseJson) => {
        try {
          if (responseJson.status == "200") {
            switch (responseJson.data.code) {
              case "1001":
                setSubmitOTPtxnId(responseJson.data.transaction_id);
                setVerifyMsg("OTP sent to User");
                setVerifyStatus("PENDING");
                setBackendPush(true);
                setVerifyTimestamp(responseJson.timestamp);
                dispatch(resetTimer());
                Analytics.trackEvent("Aadhaar|Otp|Success", {
                  userId: id,
                });
                {
                  props.type == "KYC"
                    ? navigation.navigate("KYC", {
                        screen: "AADHAAR",
                        params: {
                          screen: "Verify",
                        },
                      })
                    : navigation.navigate("AadhaarVerify");
                }
                break;
              default:
                setVerifyMsg(responseJson.data.message);
                Analytics.trackEvent("Aadhaar|Otp|Error", {
                  userId: id,
                  error: responseJson.data.message,
                });
                setVerifyStatus("ERROR");
                setBackendPush(true);
                Alert.alert("Error", responseJson.data.message);
                break;
            }
          } else if (responseJson?.error?.message) {
            setVerifyMsg(responseJson.error.message);
            Analytics.trackEvent("Aadhaar|Otp|Error", {
              userId: id,
              error: responseJson.error.message,
            });
            setVerifyStatus("ERROR");
            setBackendPush(true);
            Alert.alert("Error", responseJson.error.message);
          } else {
            setVerifyMsg(responseJson.message);
            Analytics.trackEvent("Aadhaar|Otp|Error", {
              userId: id,
              error: responseJson.message,
            });
            setVerifyStatus("ERROR");
            setBackendPush(true);
            Alert.alert("Error", responseJson.message);
          }
        } catch (error) {
          console.log("Error: ", error);
          Analytics.trackEvent("Aadhaar|Otp|Error", {
            userId: id,
            error,
          });
          setVerifyMsg(error);
          setVerifyStatus("ERROR");
          setBackendPush(true);
          Alert.alert("Error", error);
        }
      })
      .catch((error) => {
        setVerifyMsg(error);
        setVerifyStatus("ERROR");
        setBackendPush(true);
        Analytics.trackEvent("Aadhaar|Otp|Error", {
          userId: id,
          error,
        });
        Alert.alert("Error", error);
      });
  };
  return (
    <PrimaryButton
      title={loading ? "Verifying" : props.title || "Continue"}
      disabled={loading}
      loading={loading}
      onPress={() => {
        goForFetch();
      }}
    />
  );
};

export default AadhaarOtpApi;
