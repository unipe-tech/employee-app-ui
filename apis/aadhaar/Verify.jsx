import { OG_API_KEY } from "@env";
import { useEffect, useState, useRef } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Alert } from "react-native";
import { useNavigation } from "@react-navigation/core";
import {
  addData,
  addVerifyMsg,
  addVerifyStatus,
  addVerifyTimestamp,
} from "../../store/slices/aadhaarSlice";
import { KYC_AADHAAR_SUBMIT_OTP_API_URL } from "../../services/constants";
import { aadhaarBackendPush } from "../../helpers/BackendPush";
import PrimaryButton from "../../components/atoms/PrimaryButton";
import Analytics from "appcenter-analytics";
import { setValue } from "../../helpers/SetRefValue";

const AadhaarVerifyApi = (props) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const [loading, setLoading] = useState(false);

  const unipeEmployeeId = useSelector((state) => state.auth.unipeEmployeeId);
  const token = useSelector((state) => state.auth.token);
  const submitOTPtxnId = useSelector((state) => state.aadhaar.submitOTPtxnId);

  const aadhaarSlice = useSelector((state) => state.aadhaar);
  const data = useRef(aadhaarSlice?.data);
  const verifyMsg = useRef(aadhaarSlice?.verifyMsg);
  const verifyStatus = useRef(aadhaarSlice?.verifyStatus);
  const verifyTimestamp = useRef(aadhaarSlice?.verifyTimestamp);

  useEffect(() => {
    dispatch(addData(data.current));
  }, [data]);

  useEffect(() => {
    dispatch(addVerifyMsg(verifyMsg.current));
  }, [verifyMsg]);

  useEffect(() => {
    dispatch(addVerifyStatus(verifyStatus.current));
  }, [verifyStatus]);

  useEffect(() => {
    dispatch(addVerifyTimestamp(verifyTimestamp.current));
  }, [verifyTimestamp]);

  const backendPush = (props) => {
    console.log("AadhaarVerifyApi aadhaarSlice: ", aadhaarSlice);
    setValue({ ref: data, value: props.data });
    setValue({ ref: verifyMsg, value: props.verifyMsg });
    setValue({ ref: verifyStatus, value: props.verifyStatus });
    setValue({ ref: verifyTimestamp, value: props.verifyTimestamp });
    aadhaarBackendPush({
      data: {
        unipeEmployeeId: unipeEmployeeId,
        data: props.data,
        number: aadhaarSlice?.number,
        verifyMsg: props.verifyMsg,
        verifyStatus: props.verifyStatus,
        verifyTimestamp: props.verifyTimestamp,
      },
      token: token,
    });
    setLoading(false);
  };

  const goForFetch = () => {
    setLoading(true);

    const data = {
      otp: props.data.otp,
      include_xml: props.data.include_xml,
      share_code: props.data.share_code,
      transaction_id: submitOTPtxnId,
    };

    const options = {
      method: "POST",
      headers: {
        "X-Auth-Type": "API-Key",
        "X-API-Key": OG_API_KEY,
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    };

    fetch(KYC_AADHAAR_SUBMIT_OTP_API_URL, options)
      .then((response) => response.json())
      .then((responseJson) => {
        console.log("responseJson: ", responseJson);
        try {
          if (responseJson["status"] == "200") {
            switch (responseJson["data"]["code"]) {
              case "1002":
                const names = [
                  "house",
                  "street",
                  "locality",
                  "sub_district",
                  "district",
                  "state",
                  "pincode",
                ];
                responseJson["data"]["aadhaar_data"]["address"] = names
                  .filter((k) => responseJson["data"]["aadhaar_data"][k])
                  .map((k) => responseJson["data"]["aadhaar_data"][k])
                  .join(", ");
                console.log("AADHAAR fetched data: ", responseJson);
                backendPush({
                  data: responseJson["data"]["aadhaar_data"],
                  verifyMsg: "OTP validated by User",
                  verifyStatus: "PENDING",
                  verifyTimestamp: responseJson["timestamp"],
                });
                Analytics.trackEvent("Aadhaar|Verify|Success", {
                  unipeEmployeeId: unipeEmployeeId,
                });
                props.setVerified(true);
                {
                  props.type == "KYC"
                    ? navigation.navigate("KYC", {
                        screen: "AADHAAR",
                        params: {
                          screen: "Confirm",
                        },
                      })
                    : navigation.navigate("AadhaarConfirm");
                }
                break;
              default:
                backendPush({
                  data: data,
                  verifyMsg: responseJson["data"]["message"],
                  verifyStatus: "ERROR",
                  verifyTimestamp: verifyTimestamp.current,
                });
                Alert.alert("Error", responseJson["data"]["message"]);
                Analytics.trackEvent("Aadhaar|Verify|Error", {
                  unipeEmployeeId: unipeEmployeeId,
                  error: responseJson["data"]["message"],
                });
            }
          } else if (responseJson?.error?.message) {
            backendPush({
              data: data,
              verifyMsg: responseJson["error"]["message"],
              verifyStatus: "ERROR",
              verifyTimestamp: verifyTimestamp.current,
            });
            Alert.alert("Error", responseJson["error"]["message"]);
            Analytics.trackEvent("Aadhaar|Verify|Error", {
              unipeEmployeeId: unipeEmployeeId,
              error: responseJson["error"]["message"],
            });
          } else {
            backendPush({
              data: data,
              verifyMsg: responseJson["message"],
              verifyStatus: "ERROR",
              verifyTimestamp: verifyTimestamp.current,
            });
            Alert.alert("Error", responseJson["message"]);
            Analytics.trackEvent("Aadhaar|Verify|Error", {
              unipeEmployeeId: unipeEmployeeId,
              error: responseJson["message"],
            });
          }
        } catch (error) {
          backendPush({
            data: data,
            verifyMsg: error.toString(),
            verifyStatus: "ERROR",
            verifyTimestamp: verifyTimestamp.current,
          });
          Alert.alert("Error", error.toString());
          Analytics.trackEvent("Aadhaar|Verify|Error", {
            unipeEmployeeId: unipeEmployeeId,
            error: error.toString(),
          });
        }
      })
      .catch((error) => {
        backendPush({
          data: data,
          verifyMsg: error.toString(),
          verifyStatus: "ERROR",
          verifyTimestamp: verifyTimestamp.current,
        });
        Alert.alert("Error", error.toString());
        Analytics.trackEvent("Aadhaar|Verify|Error", {
          unipeEmployeeId: unipeEmployeeId,
          error: error.toString(),
        });
      });
  };

  return (
    <PrimaryButton
      accessibilityLabel={"AadhaarVerifyBtn"}
      title={loading ? "Verifying" : "Continue"}
      disabled={props.disabled}
      loading={loading}
      onPress={() => {
        goForFetch();
      }}
    />
  );
};

export default AadhaarVerifyApi;
