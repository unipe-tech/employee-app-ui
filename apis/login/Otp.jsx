import { useNavigation } from "@react-navigation/core";
import Analytics from "appcenter-analytics";
import { useEffect, useState } from "react";
import { Alert, Text } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import PrimaryButton from "../../components/atoms/PrimaryButton";
import { putBackendData } from "../../services/employees/employeeServices";
import { sendSmsVerification } from "../../services/otp/Gupshup/services";
import {
  addOnboarded,
  addToken,
  addUnipeEmployeeId
} from "../../store/slices/authSlice";
import { resetTimer } from "../../store/slices/timerSlice";
import { styles } from "../../styles";

const Otp = ({ disabled, type }) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const authSlice = useSelector((state) => state.auth);
  const [loading, setLoading] = useState(false);
  const [onboarded, setOnboarded] = useState(authSlice?.onboarded);
  const phoneNumber = authSlice?.phoneNumber;
  const [token, setToken] = useState(authSlice?.token);
  const [unipeEmployeeId, setUnipeEmployeeId] = useState(
    authSlice?.unipeEmployeeId
  );

  useEffect(() => {
    dispatch(addToken(token));
  }, [token]);

  useEffect(() => {
    dispatch(addOnboarded(onboarded));
  }, [onboarded]);

  useEffect(() => {
    dispatch(addUnipeEmployeeId(unipeEmployeeId));
  }, [unipeEmployeeId]);

  const SendOtp = (type) => {
    if (type === "login") {
      setLoading(true);
      dispatch(resetTimer());
      var fullPhoneNumber = `+91${phoneNumber}`;
      putBackendData({
        data: { number: fullPhoneNumber },
        xpath: "mobile",
        token: token,
      })
        .then((res) => {
          console.log("LoginScreen res.data: ", res.data);
          if (res.data.status === 200) {
            setOnboarded(res.data.body.onboarded);
            setToken(res.data.body.token);
            setUnipeEmployeeId(res.data.body.unipeEmployeeId);
            sendSmsVerification(phoneNumber)
              .then((result) => {
                console.log("sendSmsVerification result: ", result);
                if (result["response"]["status"] === "success") {
                  setLoading(false);
                  Analytics.trackEvent("LoginScreen|SendSms|Success", {
                    unipeEmployeeId: unipeEmployeeId,
                  });
                  navigation.navigate("Otp");
                } else {
                  setLoading(false);
                  Alert.alert(
                    result["response"]["status"],
                    result["response"]["details"]
                  );
                  Analytics.trackEvent("LoginScreen|SendSms|Error", {
                    unipeEmployeeId: unipeEmployeeId,
                    error: result["response"]["details"],
                  });
                }
              })
              .catch((error) => {
                console.log("sendSmsVerification result: ", error.toString());
                setLoading(false);
                Alert("Error", error.toString());
                Analytics.trackEvent("LoginScreen|SendSms|Error", {
                  unipeEmployeeId: unipeEmployeeId,
                  error: error.toString(),
                });
              });
            Analytics.trackEvent(`LoginScreen|SignIn|Success`, {
              unipeEmployeeId: res.data.body.id,
            });
          } else {
            setLoading(false);
            Alert.alert("Error", res.data["message"]);
            Analytics.trackEvent("LoginScreen|SignIn|Error", {
              phoneNumber: phoneNumber,
              error: res.data["message"],
            });
          }
        })
        .catch((error) => {
          console.log("LoginScreen res.data: ", error.toString());
          setLoading(false);
          Analytics.trackEvent("LoginScreen|SignIn|Error", {
            phoneNumber: phoneNumber,
            error: error.toString(),
          });
        });
    } else {
      sendSmsVerification(phoneNumber)
        .then((res) => {
          if (res["response"]["status"] === "success") {
            dispatch(resetTimer());
            Analytics.trackEvent("OTPScreen|SendSms|Success", {
              unipeEmployeeId: unipeEmployeeId,
            });
            Alert.alert("OTP resent successfully");
          } else {
            Analytics.trackEvent("OTPScreen|SendSms|Error", {
              unipeEmployeeId: unipeEmployeeId,
              error: res["response"]["details"],
            });
            Alert.alert(res["response"]["status"], res["response"]["details"]);
          }
        })
        .catch((error) => {
          console.log(error.toString());
          Analytics.trackEvent("OTPScreen|SendSms|Error", {
            unipeEmployeeId: unipeEmployeeId,
            error: error.toString(),
          });
          Alert.alert("Error", error.toString());
        });
    }
  };

  return (
    <>
      {type === "login" ? (
        <PrimaryButton
          title="Sign In"
          onPress={() => {
            SendOtp(type);
          }}
          disabled={disabled}
          loading={loading}
        />
      ) : (
        <Text
          style={styles.resendText}
          onPress={() => {
            SendOtp(type);
          }}
        >
          Resend
        </Text>
      )}
    </>
  );
};

export default Otp;
