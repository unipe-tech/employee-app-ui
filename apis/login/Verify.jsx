import { useNavigation } from "@react-navigation/core";
import Analytics from "appcenter-analytics";
import { Alert } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import PrimaryButton from "../../components/atoms/PrimaryButton";
import { checkVerification } from "../../services/otp/Gupshup/services";
import { resetTimer } from "../../store/slices/timerSlice";

const Verify = ({ disabled, otp }) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();
  const phoneNumber = useSelector((state) => state.auth.phoneNumber);
  const onboarded = useSelector((state) => state.auth.onboarded);
  const unipeEmployeeId = useSelector((state) => state.auth.unipeEmployeeId);

  const checkOtp = () => {
    checkVerification(phoneNumber, otp)
      .then((res) => {
        console.log("res: ", res);
        if (res["response"]["status"] === "success") {
          if (onboarded) {
            navigation.navigate("BackendSync", {
              destination: "HomeStack",
            });
          } else {
            navigation.navigate("BackendSync", {
              destination: "Welcome",
            });
          }
          dispatch(resetTimer());
          Analytics.trackEvent("OTPScreen|Check|Success", {
            unipeEmployeeId: unipeEmployeeId,
            error: res["response"]["details"],
          });
        } else {
          Alert.alert(res["response"]["status"], res["response"]["details"]);
          Analytics.trackEvent("OTPScreen|Check|Error", {
            unipeEmployeeId: unipeEmployeeId,
            error: res["response"]["details"],
          });
        }
      })
      .catch((error) => {
        console.log(error.toString());
        Alert.alert("Error", error.toString());
        Analytics.trackEvent("OTPScreen|Check|Error", {
          unipeEmployeeId: unipeEmployeeId,
          error: error.toString(),
        });
      });
  };

  return (
    <PrimaryButton
      title="Verify"
      disabled={disabled}
      onPress={() => {
        checkOtp();
      }}
    />
  );
};

export default Verify;
