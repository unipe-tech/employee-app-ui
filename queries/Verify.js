import { useQuery, useMutation, QueryClient } from "@tanstack/react-query";
import { OG_API_KEY } from "@env";
import { KYC_PAN_VERIFY_API_URL } from "../services/constants";

export const queryClient = new QueryClient();

export const UsePOSTVerify = ({data, url}) => {
    const {status, ...query} = useQuery(['PostPAN', {data, url}], async() => {
      console.log("data from React Query", data)
      return await fetch(url, {
          method: "POST",
          headers: {
            "X-Auth-Type": "API-Key",
            "X-API-Key": OG_API_KEY,
            "Content-Type": "application/json",
          },
          body: JSON.stringify(data),
        }).then(response => {
          return response.json()
        }).catch(error => {
          return error;
        })
  
    })
    return {status, ...query}
}

export const UseAddData = () => {
 const mutation = useMutation(
    async(data, url) => {
      return await fetch(url, {
        method: "POST",
        headers: {
          "X-Auth-Type": "API-Key",
          "X-API-Key": OG_API_KEY,
          "Content-Type": "application/json",
        },
        body: JSON.stringify(data),
      }).then(response => {
        return response.json()
      }).catch(error => {
        return error;
      })
  }, {
    onSuccess: () => {
      queryClient.invalidateQueries("addData");
    }
  })

  return mutation;

}