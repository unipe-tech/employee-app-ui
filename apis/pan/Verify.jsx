import { OG_API_KEY } from "@env";
import { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Alert } from "react-native";
import { useNavigation } from "@react-navigation/core";
import {
  addData,
  addVerifyMsg,
  addVerifyStatus,
  addVerifyTimestamp,
} from "../../store/slices/panSlice";
import { KYC_PAN_VERIFY_API_URL } from "../../services/constants";
import { panBackendPush } from "../../helpers/BackendPush";
import PrimaryButton from "../../components/PrimaryButton";
import Analytics from "appcenter-analytics";
import { UseAddData,  UsePOSTVerify } from "../../queries/Verify";

const PanVerifyApi = (props) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();

  const [loading, setLoading] = useState(false);
  const [backendPush, setBackendPush] = useState(false);

  const [responseJson, setResponseJson] = useState({})
  const id = useSelector((state) => state.auth.id);
  const panSlice = useSelector((state) => state.pan);
  const [data, setData] = useState(panSlice?.data);
  const [verifyMsg, setVerifyMsg] = useState(panSlice?.verifyMsg);
  const [verifyStatus, setVerifyStatus] = useState(panSlice?.verifyStatus);
  const [verifyTimestamp, setVerifyTimestamp] = useState(
    panSlice?.verifyTimestamp
  );

  useEffect(() => {
    dispatch(addData(data));
  }, [data]);

  useEffect(() => {
    dispatch(addVerifyMsg(verifyMsg));
  }, [verifyMsg]);

  useEffect(() => {
    dispatch(addVerifyStatus(verifyStatus));
  }, [verifyStatus]);

  useEffect(() => {
    dispatch(addVerifyTimestamp(verifyTimestamp));
  }, [verifyTimestamp]);

  useEffect(() => {
    console.log("PanVerifyApi backendPush panSlice: ", backendPush, panSlice);
    if (backendPush) {
      panBackendPush({
        id: id,
        data: data,
        number: panSlice?.number,
        verifyMsg: verifyMsg,
        verifyStatus: verifyStatus,
        verifyTimestamp: verifyTimestamp,
      });
      setBackendPush(false);
    }
    setLoading(false);
  }, [backendPush]);

  // var { status, ...query } = useQuery(["panFetch",props.data], () => {
  //   const options = {
  //     method: "POST",
  //     headers: {
  //       "X-Auth-Type": "API-Key",
  //       "X-API-Key": OG_API_KEY,
  //       "Content-Type": "application/json",
  //     },
  //     body: JSON.stringify(props.data),
  //   };

  //   if (props.data.consent === "Y") {
  //     return fetch(KYC_PAN_VERIFY_API_URL, options)
  //       .then((response) => {
  //         return response.json();
  //       })
  //       .catch((error) => {
  //         console.log("Error: ", error);
  //         setVerifyMsg(error);
  //         setVerifyStatus("ERROR");
  //         Analytics.trackEvent("Pan|Verify|Error", {
  //           userId: id,
  //           error: error.toString(),
  //         });
  //         // setBackendPush(true);
  //         Alert.alert("Error", error);
  //         return error;
  //       });
  //   }
  // });

  // var { ...query} = UsePOSTVerify({data: props.data, url: KYC_PAN_VERIFY_API_URL})
  var { mutate, isLoading, isError, isIdle, isSuccess} = UseAddData()

  useEffect(() => {
    mutate( props.data, KYC_PAN_VERIFY_API_URL).then((response) => {
      console.log(response)
      setResponseJson(response)
    }).catch(console.log)
  },[])

  const goForFetch = () => {
    
    // var responseJson = query.data;
    console.log("PAN ResponseJSON: ", responseJson)
    try {
      if (responseJson["status"] == "200") {
        switch (responseJson["data"]["code"]) {
          case "1000":
            var names = ["first", "middle", "last"];
            responseJson["data"]["pan_data"]["name"] = names
              .filter((k) => responseJson["data"]["pan_data"][`${k}_name`])
              .map((k) => responseJson["data"]["pan_data"][`${k}_name`])
              .join(" ");
            console.log("PAN fetched data: ", responseJson);
            setData(responseJson["data"]["pan_data"]);
            setVerifyMsg("To be confirmed by User");
            setVerifyStatus("PENDING");
            setVerifyTimestamp(responseJson["timestamp"]);
            setBackendPush(true);
            Analytics.trackEvent("Pan|Verify|Success", {
              userId: id,
            });
            {
              props.type == "KYC"
                ? navigation.navigate("KYC", {
                    screen: "PAN",
                    params: {
                      screen: "Confirm",
                    },
                  })
                : navigation.navigate("PanConfirm");
            }
            break;
          default:
            setVerifyMsg(responseJson["data"]["message"]);
            setVerifyStatus("ERROR");
            setBackendPush(true);
            Alert.alert("Error", responseJson["data"]["message"]);
            Analytics.trackEvent("Pan|Verify|Error", {
              userId: id,
              error: responseJson["data"]["message"],
            });
        }
      } else if (responseJson?.error?.message) {
        setVerifyMsg(responseJson["error"]["message"]);
        setVerifyStatus("ERROR");
        setBackendPush(true);
        Alert.alert("Error", responseJson["error"]["message"]);
        Analytics.trackEvent("Pan|Verify|Error", {
          userId: id,
          error: responseJson["error"]["message"],
        });
      } else {
        setVerifyMsg(responseJson["message"]);
        setVerifyStatus("ERROR");
        setBackendPush(true);
        Alert.alert("Error", responseJson["message"]);
        Analytics.trackEvent("Pan|Verify|Error", {
          userId: id,
          error: responseJson["message"],
        });
      }
      } catch (error){
        console.log("Error casda: ", error);
        setVerifyMsg(error);
        setVerifyStatus("ERROR");
        setBackendPush(true);
        Alert.alert("Error", error);
        Analytics.trackEvent("Pan|Verify|Error", {
          userId: id,
          error: error,
        });
      }
      
    }
  


  return (
    <PrimaryButton
      title={loading ? "Verifying" : "Continue"}
      disabled={props.disabled}
      loading={loading}
      onPress={() => {
        goForFetch();
      }}
    />
  );
};

export default PanVerifyApi;
