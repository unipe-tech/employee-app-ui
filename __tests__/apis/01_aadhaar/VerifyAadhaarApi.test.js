import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { Provider } from "react-redux";
import { store } from "../../../store/store";
import { shallow, mount } from "enzyme";
import Otp from "../../../apis/aadhaar/Otp";
import { render, fireEvent } from "@testing-library/react-native";
import { Button } from "@react-native-material/core";
import "isomorphic-fetch";
import { act } from "react-test-renderer";
import ApiView from "../../../apis/ApiView";
import Verify from "../../../apis/aadhaar/Verify";
import mockStore from "redux-mock-store";

describe("AadhaarOTP API", () => {
  describe("UI rendering test", () => {
    it("testing UI rendering", () => {
      const tree = shallow(
        <Provider store={store}>
          <NavigationContainer>
            <Otp
              url="http://unipeapi.herokuapp.com/aadhaar-api/boson/submit-otp"
              data={{
                otp: 345579,
                include_xml: true,
                share_code: "1234",
              }}
              disabled={true}
            />
          </NavigationContainer>
        </Provider>
      );
      expect(tree).toMatchSnapshot();
    }, 8000);

    it("testing OTP component", () => {
      const goForFetch = jest.fn({
        fetchUrl:
          "http://unipeapi.herokuapp.com/aadhaar-api/boson/generate-otp",
        postData: { aadhaar_number: "123456789101", consent: "Y" },
      });
      const navigate = jest.fn();
      const store = mockStore();

      const component = render(
        <Provider store={store}>
          <NavigationContainer>
            <Otp
              url="http://unipeapi.herokuapp.com/aadhaar-api/boson/submit-otp/"
              data={{
                otp: 345579,
                include_xml: true,
                share_code: "1234",
              }}
              disabled={false}
              goForFetch={goForFetch}
              navigation={{ navigate }}
            />
          </NavigationContainer>
        </Provider>
      );
      const wrapper = shallow(
        <Provider store={store}>
          <NavigationContainer>
            <Verify
              url="http://unipeapi.herokuapp.com/aadhaar-api/boson/submit-otp/"
              data={{
                otp: 345579,
                include_xml: true,
                share_code: "1234",
              }}
              disabled={false}
              goForFetch={goForFetch}
              navigation={{ navigate }}
            />
          </NavigationContainer>
        </Provider>
      );
      // fireEvent.press(component.getByTestId("apiButton"));
      // wrapper.find(ApiView).simulate("click");
      // wrapper
      //   .find(<Button />)
      //   .props()
      //   .onClick();
      // expect(
      //   wrapper.findWhere((node) => node.prop("testID") === "apiView")
      // ).toExist();

      act(async () => {
        await expect(goForFetch.mock.calls.length).toEqual(1);
        await expect(goForFetch).toHaveBeenCalledTimes(1);
        await expect(navigate).toHaveBeenCalledWith("AadhaarConfirm");
      });
    });
  });
  //   describe("State tests", () => {
  //     let wrapper;
  //     let component;
  //     const setState = jest.fn();
  //     const useStateSpy = jest.spyOn(React, "useState");
  //     useStateSpy.mockImplementation((init) => [init, setState]);

  //     beforeEach(() => {
  //       const goForFetch = jest.fn();
  //       const navigate = jest.fn();
  //       wrapper = shallow(
  //         <Provider store={store}>
  //           <NavigationContainer>
  //             <Otp
  //               url="http://unipeapi.herokuapp.com/aadhaar-api/boson/generate-otp/200/1001"
  //               data={{ aadhaar_number: "123456789101", consent: "Y" }}
  //               disabled={false}
  //               goForFetch={goForFetch}
  //               navigation={{ navigate }}
  //             />
  //           </NavigationContainer>
  //         </Provider>
  //       );
  //       component = render(
  //         <Provider store={store}>
  //           <NavigationContainer>
  //             <Otp
  //               url="http://unipeapi.herokuapp.com/aadhaar-api/boson/generate-otp/200/1001"
  //               data={{ aadhaar_number: "123456789101", consent: "Y" }}
  //               disabled={false}
  //               goForFetch={goForFetch}
  //               navigation={{ navigate }}
  //             />
  //           </NavigationContainer>
  //         </Provider>
  //       );
  //     });

  //     afterEach(() => {
  //       jest.clearAllMocks();
  //     });

  //     describe("Should set Verify Message", () => {
  //       it("call setVerifyMessage and set verify message to sent", () => {
  //         expect(wrapper.find({ testID: "apiButton" })).toBeDefined();
  //         // wrapper.find({ testID: "apiButton" }).first().simulate("click");
  //         const goForFetch = jest.fn();
  //         fireEvent.press(component.getByText("Continue"));
  //         // expect(goForFetch).toBeCalled();
  //         // act(async () => {
  //         //   await expect(goForFetch).toHaveBeenCalledWith({
  //         //     fetchUrl:
  //         //       "http://unipeapi.herokuapp.com/aadhaar-api/boson/generate-otp/200/1001",
  //         //     postData: { aadhaar_number: "123456789101", consent: "Y" },
  //         //   });
  //         // });
  //         expect(setState).toHaveBeenCalled();
  //       });
  //     });
  //   });
});
