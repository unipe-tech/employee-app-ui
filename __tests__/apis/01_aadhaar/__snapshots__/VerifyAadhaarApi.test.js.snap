// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AadhaarOTP API UI rendering test testing UI rendering 1`] = `
<ContextProvider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <Otp
      data={
        Object {
          "include_xml": true,
          "otp": 345579,
          "share_code": "1234",
        }
      }
      disabled={true}
      url="http://unipeapi.herokuapp.com/aadhaar-api/boson/submit-otp"
    />
  </ForwardRef(NavigationContainerInner)>
</ContextProvider>
`;
